src/tuning/kernels/xgemm.cpp

```diff
@@ -190,10 +196,49 @@ void StartVariation(int argc, char *argv[]) {
   }
 }
 
+
 // Main function (not within the clblast namespace)
 int main(int argc, char *argv[]) {
+
+int num_of_strategies = DEFAULT_NUM_OF_STRATEGY;
+
+if(const char* env_p = std::getenv("CK_TUNER_NUM_OF_STRATEGIES"))
+{ 
+  num_of_strategies = atoi(env_p);
+  printf("CK_TUNER_NUM_OF_STRATEGIES is: %s\n", env_p );
+}
+else
+{
+  printf("CK_TUNER_NUM_OF_STRATEGIES is not defined\n");     
+}
+
+printf("num_of_strategies : %d\n",num_of_strategies );
+
+if( DEFAULT_NUM_OF_STRATEGY != num_of_strategies )
+{
+  //FACCIO PRIMA LA FULL SEARCH
   StartVariation<1>(argc, argv);
-  StartVariation<2>(argc, argv);
+}
+
+  auto command_line_args = clblast::RetrieveCommandLineArguments(argc, argv);
+  auto help = std::string{"* Options given/available:\n"};
+  auto localtStrategy = clblast::GetArgument(command_line_args, help, clblast::tStrategy, DEFAULT_STRATEGY); 
+  switch(localtStrategy)
+  {
+    case FULL_SEARCH_STRATEGY: 
+      StartVariation<1>(argc, argv);
+      break;
+    case RANDOM_SEARCH_STRATEGY: 
+      StartVariation<2>(argc, argv);
+      break;
+    case PSO_STRATEGY:
+      StartVariation<3>(argc, argv);
+      break;
+    case DVDT_STRATEGY:
+      StartVariation<2>(argc, argv);
+      break;
+  }
+
```

Comment:
Can we not keep it a bit simpler the way it was? And just replace random search by PSO and perhaps add an option to switch it back or so? This all seems overly complicated to me.